src/util/input-types.ts

```diff
@@ -0,0 +1,10 @@
+import {Observable, ObservableInput} from '../Observable';
+
+export type _mapProject<T, R> = (value: T, index: number) => R;
```

Comment:
It can be, in that case those new PR should be checked in prior to this though.

- is `util' feasible place for these? (I'm also not sure)
- as above comment, does `_` needed?